s3prl/upstream/multires_hubert/expert.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) Facebook, Inc. All Rights Reserved
+
+# -*- coding: utf-8 -*- #
+"""*********************************************************************************************"""
+#   FileName     [ upstream/multires_hubert/expert.py ]
+#   Synopsis     [ the Multiresolution HuBERT wrapper ]
+#   Author       [ Jiatong Shi ]
+"""*********************************************************************************************"""
+
+import logging
+from pathlib import Path
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch.nn.utils.rnn import pad_sequence
+
+from ..interfaces import UpstreamBase
+from .convert import load_converted_model
+
+SAMPLE_RATE = 16000
+EXAMPLE_SEC = 5
+
+logger = logging.getLogger(__name__)
+
+
+def upsample(x, upsample_rate):
+    return x.repeat_interleave(upsample_rate, dim=1)
+
+
+class UpstreamExpert(UpstreamBase):
+    def __init__(self, ckpt, legacy=True, **kwargs):
+        super().__init__(**kwargs)
+        model, task_cfg = load_converted_model(ckpt)
+        self.model = model
+        self.task_cfg = task_cfg
+        self.final_factors = []
+
+        self.num_res = self.model.label_nums
+        logger.info("num res: {}".format(self.num_res))
+        # decide ratios
+        if legacy:
```

Comment:
There is no entry with `legacy=False`, what does this option mean in this case?

Reply:
My bad, will fix it soon